templates/makefile-c++/makefile

```diff
@@ -0,0 +1,60 @@
+# Clean makefile based on https://stackoverflow.com/questions/30573481/how-to-write-a-makefile-with-separate-source-and-header-directories
+# Modified to use g++. 
+
+##############################
+#	USER DEFINED VARIABLES	 #
+##############################
+# Set the name of the target executable
+PROJECT_NAME := main
+# Add any arguments to the `make run` routine, if applicable 
+RUN_ARGS := 
+
+##############################
+#      STATIC VARIABLES      #
+##############################
+SRC_DIR := src
+OBJ_DIR := obj
+BIN_DIR := bin
+
+EXE := $(BIN_DIR)/$(PROJECT_NAME)
+SRC := $(wildcard $(SRC_DIR)/*.cpp)
+OBJ := $(SRC:$(SRC_DIR)/%.cpp=$(OBJ_DIR)/%.o)
+
+##############################
+#       COMPILER SETUP       #
+##############################
+CXX=g++
+CPPFLAGS := -Iinclude -MMD -MP
+CFLAGS   := -Wall -g
+LDFLAGS  := -Llib 
```

Comment:
Question: what is `lib` referring to? The `-L` flag followed by a path will pull in shared libraries in the specified directory. Not sure what shared libs are being referred to here but I am no linker wiz. Tried the Makefile on Linux and it works with or without this flag. Note, by shared libraries, I mean the pre-compiled stuff like `.dll`, `.so`, `.a` etc.

Reply:
> Question: what is `lib` referring to? The `-L` flag followed by a path will pull in shared libraries in the specified directory. Not sure what shared libs are being referred to here but I am no linker wiz. Tried the Makefile on Linux and it works with or without this flag. Note, by shared libraries, I mean the pre-compiled stuff like `.dll`, `.so`, `.a` etc.

lib refers to the `lib` folder in the template that is currently not present but similar to the `-Iinclude` flag where `include` is the folder for headers. The lib folder is for shared libraries (.dll, .a, .so type files, I presume) that need to be linked but we do not have source code for. Not sure we will ever use this, but I wanted it present for completeness. 

I think I will variable out this path so that library paths can be specified.

Reply:
Good call.